saleor/warehouse/management.py

```diff
@@ -33,39 +36,54 @@ def allocate_stocks(order_lines_info: Iterable["OrderLineData"], country_code: s
         return
 
     variants = [line_info.variant for line_info in order_lines_info]
+    if None in variants:
+        raise ValidationError(
+            {
+                "variant": ValidationError(
+                    "Missing variant in order line. Cannot allocate the stock",
+                    code=StockErrorCode.INVALID.value,
+                )
```

Comment:
Missing variant shouldn't happen here, I assume this is also due to a race condition when a customer tries to buy a variant, that is suddenly removed from the system?

Maybe it would be better to change the line above:
```
variants = [line_info.variant for line_info in order_lines_info if line_info.variant]
```
And only create allocations for existing variants? What do you think @fowczarek?

Reply:
If we want to raise an error I'm not sure if this should be a validation error. A missing variant in this place is the same as InsufficientStock error from the user's perspective.

Reply:
I'm agree with that missing variant shouldn't happen here. I added this check to be sure because OrderLineInfo have variant as optional field.

Reply:
So we should raise `InsufficientStock` in this case, since creating allocation can throw this error anyway, and it should be handled automatically at the mutation level. See how insufficient stock errors are returned in `_create_allocations`. Also we need to add a test for this case.

Reply:
I removed this validation. This should be discussed and fixed in other PR.